src/App.tsx

```diff
@@ -14,210 +30,109 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  useEffect(() => {
+    async function todosFromServer(userId: number | undefined) {
+      try {
+        const visibleTodos = getTodos(userId);
+
+        setTodos(await visibleTodos);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    }
+
+    if (!user) {
+      return;
+    }
+
+    todosFromServer(user.id);
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.All:
+          return todo;
+
+        case FilterBy.Active:
+          return !todo.completed;
+
+        case FilterBy.Completed:
+          return todo.completed;
+
+        default:
+          return null;
+      }
+    });
+  }, [todos, filterBy]);
+
+  const handleTodos = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setErrorMessage('Title can`t be empty');
+
+      return;
+    }
+
+    setAdding(true);
+
+    try {
+      const newTodo = await createTodos(user?.id, title);
+
+      setTodos([...todos, newTodo]);
+    } catch {
+      setErrorMessage('Can\'t be added');
+    }
+
+    setTitle('');
```

Comment:
to previous coment: if you will change it, so you need some default return, for example `return setTitle('');` and switch position with `setAdding(false);`

Reply:
@Mejeezz what created by setState() functions returns? => yourAnswer
``Why do @anyakrainiuk should return ${yourAnswer} here?``